hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/UdtObjectToJsonFunction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.expression;
+
+import com.hazelcast.core.HazelcastJsonValue;
+import com.hazelcast.jet.sql.impl.JetSqlSerializerHook;
+import com.hazelcast.jet.sql.impl.expression.json.JsonCreationUtil;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.query.impl.getters.Extractors;
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.SqlErrorCode;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.expression.UniExpressionWithType;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.type.QueryDataType;
+
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.newSetFromMap;
+
+public class UdtObjectToJsonFunction extends UniExpressionWithType<HazelcastJsonValue> implements IdentifiedDataSerializable {
+
+    public UdtObjectToJsonFunction() { }
+
+    private UdtObjectToJsonFunction(Expression<?> operand) {
+        super(operand, QueryDataType.JSON);
+    }
+
+    public static UdtObjectToJsonFunction create(Expression<?> operand) {
+        return new UdtObjectToJsonFunction(operand);
+    }
+
+    @Override
+    public HazelcastJsonValue eval(final Row row, final ExpressionEvalContext context) {
+        final Object obj = this.operand.eval(row, context);
+        final QueryDataType queryDataType = operand.getType();
+
+        if (obj == null) {
+            return null;
+        }
+
+        final Extractors extractors = Extractors.newBuilder(context.getSerializationService()).build();
```

Comment:
The instance should be cached in a field. It has a getter cache for performance, creating new extractors each time will make the cache useless.